src/client/dfuse/ops/setattr.c

```diff
@@ -60,13 +60,22 @@ dfuse_cb_setattr(fuse_req_t req, struct dfuse_inode_entry *ie,
 				attr->st_size);
 		to_set &= ~(FUSE_SET_ATTR_SIZE);
 		dfs_flags |= DFS_SET_ATTR_SIZE;
+		if (ie->ie_dfs->dfs_attr_timeout > 0 &&
+		    ie->ie_stat.st_size == 0 && attr->st_size > 0) {
+			DFUSE_TRA_DEBUG(ie, "truncating 0-size file");
+			ie->ie_truncated = true;
+			ie->ie_start_off = 0;
+			ie->ie_end_off = 0;
+			ie->ie_stat.st_size = attr->st_size;
+		} else {
+			ie->ie_truncated = false;
+		}
```

Comment:
When write back is set, you also get FUSE_SET_ATTR_CTIME. Shouldn't we handle them too here?

Reply:
Ah, I was thinking the CTIME was a result of the allow-root, yes it makes more sense if it's related to writeback.

Reply:
to be fixed.